src/modules/household/hooks/usePerformJoinHousehold.tsx

```diff
@@ -25,6 +28,12 @@ export function usePerformJoinHousehold({
       if (data.joinHousehold) {
         setJoinedHousehold(data.joinHousehold.receivingHousehold);
         setRemainingHousehold(data.joinHousehold.donorHousehold || undefined);
+
+        if (donorHouseholdId && !data.joinHousehold.donorHousehold) {
+          // There were no remaining members in the donor household, so clear it from the cache
+          cache.evict({ id: `Household:${donorHouseholdId}` });
```

Comment:
Maybe add a comment explaining: "if there _are_ remaining household members in the donor household, the donor household gets updated in the cache by the donorHousehold resolved on the mutation return value. Thats why we only need to evict from the cache explicitly if there is no donorHousehold resolved"

That wasn't immediately obvious to me (why we have this `if` condition for cache eviction)